eradiate/scenes/atmosphere/homogeneous.py

```diff
@@ -6,84 +6,105 @@
-    homogeneous medium. Scattering uses the Rayleigh phase function and the
-    Rayleigh scattering coefficient of air at standard number density (
-    see :func:`sigma_s_air`).
+    This class builds an atmosphere consisting of a homogeneous medium.
+    Scattering uses the Rayleigh phase function.
 
     See :class:`~eradiate.scenes.atmosphere.base.Atmosphere` for undocumented
     members.
 
     .. rubric:: Constructor arguments / instance attributes
 
-    ``sigma_s`` (float or "auto"):
+    ``sigma_s`` (:class:`~eradiate.scenes.spectra.Spectrum` or "auto"):
         Atmosphere scattering coefficient value. If set to ``"auto"``,
         the scattering coefficient will be computed based on the current
         operational mode configuration using the :func:`sigma_s_air`
         function. Default: ``"auto"``.
 
-        Unit-enabled field (default unit: cdu[collision_coefficient]).
+        Can be initialised with a dictionary processed by
+        :class:`.SceneElementFactory`.
+
+    ``sigma_a`` (:class:`~eradiate.scenes.spectra.Spectrum`):
+        Atmosphere absorption coefficient value. Default: 0 (no absorption).
+
+        Can be initialised with a dictionary processed by
+        :class:`.SceneElementFactory`.
 
     """
 
-    sigma_s = attrib_quantity(
+    sigma_s = attr.ib(
         default="auto",
-        converter=converter_or_auto(converter_to_units(cdu.generator("collision_coefficient"))),
-        validator=validator_or_auto(validator_units_compatible(ureg.m ** -1), validator_is_positive),
-        units_compatible=ureg.m ** -1,
-        units_add_converter=False,
-        units_add_validator=False,
-    )  # TODO: turn into a Spectrum
+        converter=converter_or_auto(
+            Spectrum.converter("collision_coefficient")
+        ),
+        validator=validator_or_auto(
+            attr.validators.instance_of(Spectrum),
+            validator_has_quantity("collision_coefficient")
+        ),
+    )
+
+    sigma_a = attr.ib(
+        factory=lambda: UniformCollisionCoefficientSpectrum(value=0.),
+        converter=Spectrum.converter("collision_coefficient"),
+        validator=[attr.validators.instance_of(Spectrum),
+                   validator_has_quantity("collision_coefficient")]
+    )
 
     @property
     def kernel_height(self):
         if self.height == "auto":
-            height = ureg.Quantity(100, "km")
+            return ureg.Quantity(100, "km")
         else:
-            height = self.height
+            return self.height
 
-        return height.to(kdu.get("length"))
+    @property
+    def kernel_width(self):
+        if self.width == "auto":
+            return 10. / self._sigma_s.value
+        else:
+            return self.width
 
     @property
     def _albedo(self):
         """Return albedo."""
-        return 1.
+        return UniformAlbedoSpectrum(
+            value=self._sigma_s.value / (self._sigma_s.value + self.sigma_a.value)
```

Comment:
I cannot try that at the moment, but what happens if one creates a homogeneous atmosphere with a zero scattering coefficient and a zero absorption coefficient? Do we check somewhere that the extinction coefficient is strictly positive?

Reply:
No, but it's actually quite complicated. I currently don't have a reliable way (meaning something which won't break as soon as you'll start using a different spectrum type) to test for this upon object initialisation: we don't have spectrum arithmetics. So for the moment, I'll just leave this reponsibility to the user.